src/z3c/form/browser/widget.rst

```diff
@@ -0,0 +1,45 @@
+Widget base classes
+===================
+
+HTMLFormElement
+---------------
+The widget base class.
+::
+
+  >>> from z3c.form.browser.widget import HTMLFormElement
+  >>> form = HTMLFormElement()
+
+
+addClass
+........
+
+Widgets based on :code:`HTMLFormElement` also have the :code:`addClass` method which can be used to add CSS classes to the widget.
+
+The :code:`klass` attribute is used because :code:`class` is a reserved keyword in Python.
+It's empty per default::
+
+  >>> form = HTMLFormElement()
+  >>> form.klass
+
+
+After adding a class it shows up in :code:`klass`::
+
+  >>> form.addClass("my-css-class")
+  >>> form.klass
+  'my-css-class'
+
+
+:code:`addClass` prevents adding the same class twice::
+
+  >>> form.addClass("my-css-class")
+  >>> form.klass
+  'my-css-class'
+
+  >>> form.addClass("another-class")
+  >>> form.klass
+  'my-css-class another-class'
+
+  >>> form.addClass("another-class third-class")
```

Comment:
What would be the result when you would have added here `third-class another-class`?
Is it `'my-css-class another-class third-class'` like before or `'my-css-class third-class another-class'`? How did the previous implementation behave?

Reply:
I‌ updated the test with your example. The result is as the correct:
```
'my-css-class another-class third-class'
```

The previous implementation did not fully support a string with whitespace separated classes. The test fails with:

```
Failure in test /home/_thet/data/dev/sprints/plone6ui/REPOS/buildout.coredev/src/z3c.form/src/z3c/form/tests/../browser/widget.rst
Failed doctest test for widget.rst
  File "/home/_thet/data/dev/sprints/plone6ui/REPOS/buildout.coredev/src/z3c.form/src/z3c/form/tests/../browser/widget.rst", line 0

----------------------------------------------------------------------
File "/home/_thet/data/dev/sprints/plone6ui/REPOS/buildout.coredev/src/z3c.form/src/z3c/form/tests/../browser/widget.rst", line 43, in widget.rst
Failed example:
    form.klass
Expected:
    'my-css-class another-class third-class'
Got:
    'my-css-class another-class another-class third-class'
----------------------------------------------------------------------
File "/home/_thet/data/dev/sprints/plone6ui/REPOS/buildout.coredev/src/z3c.form/src/z3c/form/tests/../browser/widget.rst", line 50, in widget.rst
Failed example:
    form.klass
Expected:
    'my-css-class another-class third-class'
Got:
    'my-css-class another-class third-class third-class another-class'
```

But in Plone we did (incorrectly) use `addClass` with whitespace separated classes. Not a big deal because we did not add an already existing class to the widget but it shows at least that the developer expectation was different from the implementation: https://github.com/plone/plone.app.registry/blob/master/plone/app/registry/browser/edit.py#L35

Reply:
I like the adding at the end, thank you for the clarification.